File: sdk-kotlin/src/main/kotlin/org/sinou/pydia/sdk/transport/auth/credentials/LegacyPasswordCredentials.java
package org.sinou.pydia.sdk.transport.auth.credentials;

import org.sinou.pydia.sdk.api.PasswordCredentials;

public class LegacyPasswordCredentials implements PasswordCredentials {

    private final String login;
    private final String password;

    public LegacyPasswordCredentials(String login, String password) {
        this.login = login;
        this.password = password;
    }

    @Override
    public String getUsername() {
        return login;
    }

    @Override
    public String getType() {
        return TYPE_LEGACY_PASSWORD;
    }

    @Override
    public String getEncodedValue() {
        return getPassword();
    }

    @Override
    public String getPassword() {
        return password;
    }

}

File: sdk-kotlin/src/main/kotlin/org/sinou/pydia/sdk/transport/auth/credentials/LegacyPasswordCredentials.kt
package org.sinou.pydia.sdk.transport.auth.credentials

import org.sinou.pydia.sdk.api.CredentialType
import org.sinou.pydia.sdk.api.Credentials
import org.sinou.pydia.sdk.api.PasswordCredentials

class LegacyPasswordCredentials(private val login: String, private val password: String) :
    PasswordCredentials {

    override fun getUsername(): String {
        return login
    }

    override fun getType(): CredentialType {
        return CredentialType.LEGACY_PASSWORD
    }

    override fun getEncodedValue(): String {
        return getPassword()
    }

    override fun getPassword(): String {
        return password
    }
}